src/sortStudents.ts

```diff
@@ -1,16 +1,81 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
   // write your function
+  const newStude = students.map((studier) => studier);
+  let result: Student[] = [];
```

Comment:
create here the copy of the original array and mutate it.
```suggestion
  let result: Student[] = [...students];
```